### PR TITLE
feat: core accordion

### PR DESCRIPTION
**Pull request by jeeyun**
Core implementation of the Clarity accordion.

### Notes
- Core accordion doesn't manage state and therefore does not automatically close a panel if another is open, etc. This logic, along with any other logic (e.g. intercept when user expands / closes a panel) can be implemented on the application side
- Slide animation for expand / close (animating from `height: 0` to `height: auto`) is not included in this PR. Our current goal, if possible, is to not use javascript in our animation. If we were to use javascript to grab the content's height there are some challenges with responding to dynamic changes in accordion content's height at runtime. We'll have to revisit the animation part especially in the context of Clarity motion and how they might fit in with the rest of the components in the system.

### Reference links for expand / collapse animation
- https://developers.google.com/web/updates/2017/03/performant-expand-and-collapse
- https://jdsteinbach.com/css/holy-grail-css-animation/
- https://github.com/tweenjs/tween.js/blob/master/dist/tween.esm.js
- https://stackoverflow.com/questions/54520554/custom-element-getrootnode-closest-function-crossing-multiple-parent-shadowd

Signed-off-by: Jeeyun Lim <jeeyun.lim@gmail.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/internal/base/accordion-panel.base.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { html, LitElement } from 'lit-element';
+import { event, EventEmitter } from '../decorators/event.js';
+import { property } from '../decorators/property.js';
+export class CdsBaseAccordionPanel extends LitElement {
```
Comment (mathisscott) [inline packages/core/src/internal/base/accordion-panel.base.ts]: I'm not sure how I feel about such a sparse base class. I think this is bringing us too close to the line of inheritance versus composition – a.k.a. the murder-robot-dog problem.

What if we needed an accordion panel that was also a focus trap? With this approach, I don't see how we could do that without rewriting parts of one or the other in the same class. And in this case I feel like the accordion base class loses out because it is so limited.

Did we consider making an accordion component and extending _that_ to make things like the stepper?

File: packages/core/tokens.yml [hunk 2]
```diff
@@ -21,6 +21,10 @@ props:
     value: 'hsl(197, 68%, 65%)'
     type: 'hsl'
     category: 'global'
+  - name: selection-color
```
Comment (mathisscott) [inline packages/core/tokens.yml:24]: 👍

File: packages/core/src/accordion/accordion.stories.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      </cds-accordion-panel>
+    </cds-accordion>
+  `;
+};
+
+export const basicAccordion = () => {
+  return html`
+    <cds-accordion>
+      <cds-accordion-panel expanded>
+        <cds-accordion-header>Item 1</cds-accordion-header>
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.stories.ts]: Instead of `Item1`, `Item2` and `Item3` can we have the titles reflect what each panel is demoing?

• Expanded accordion
• Disabled accordion
• Collapsed accordion

I think it will help answer some common questions.

File: packages/core/src/accordion/accordion.stories.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        <cds-accordion-content>Content 2</cds-accordion-content>
+      </cds-accordion-panel>
+      <cds-accordion-panel>
+        <cds-accordion-header>Item 3</cds-accordion-header>
+        <cds-accordion-content>Content 3</cds-accordion-content>
+      </cds-accordion-panel>
+    </cds-accordion>
+  `;
+};
+
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.stories.ts:45]: What would an accordion that asynchronously loaded its content look like? 


As in, a user clicks on the title, the panel expands, and what?

While I would think that this is a pattern, it may make for a good demo.

File: packages/core/src/accordion/accordion.element.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @element cds-accordion
+ * @slot - Content slot for inside the accordion
+ * @cssprop --border-color
+ * @cssprop --border-width
+ * @cssprop --border-radius
+ */
+export class CdsAccordion extends LitElement {
+  render() {
+    return html` <slot name="accordion-panel"></slot> `;
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.element.ts]: Why does this slot need to be named?

File: packages/core/src/accordion/accordion.element.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Web component accordion
+ *
+ * ```typescript
+ * import '@clr/core/accordion/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-accordion>
+ *   <cds-accordion-panel expanded='panel1Expanded' (expandedChange)='panel1Expanded = $event.detail'>
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.element.ts]: this sample code looks like Angular. do you think that would confuse anyone?

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  private toggle() {
+    this.expanded = !this.expanded;
+    this.expandedChange.emit(this.expanded);
+  }
+
+  render() {
+    return html` <div class="private-host" role="group">
+      <div class="accordion-header${this.expanded ? ' open' : ''}${this.disabled ? ' disabled' : ''}">
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.ts:69]: We seem to have several approaches to boolean attributes. 

Here: `${this.disabled ? ' disabled' : ''}`

`control.element.ts, line 208`: `.disabled="${this.disabled}"`

`file.element.ts, line 45` (also line 75 here): `?disabled=${this.disabled}`

It appears that `.disabled` has won out more often than not. But I think we should make a case for which is the best way and converge around that.

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+          type="button"
+          @click="${() => this.toggle()}"
+          class="accordion-header-button"
+          ?disabled="${this.disabled}"
+          aria-disabled="${this.disabled}"
+          aria-expanded="${this.expanded}"
+        >
+          <span class="accordion-status" cds-layout="vertical align:vertical-center p-r:sm">
+            <span class="accordion-angle-wrapper">
+              <cds-icon class="accordion-angle" shape="angle" dir="right" size="14"></cds-icon>
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.ts]: Does the accordion angle change direction when open? 

`dir` is also something we are considering deprecating. i recommend using `direction` instead. same values, longer name. 😁 

Also, can people put their own icon here? How could we enable them to do that?

Comment (mathisscott) [reply packages/core/src/accordion/accordion-panel.element.ts]: I think we are also missing some aria here for a11y...

File: packages/core/src/accordion/entrypoint.tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "../../tsconfig.lib.json",
+  "compilerOptions": {
+    "composite": true
+  },
+  "references": [{ "path": "../internal/entrypoint.tsconfig.json" }]
```
Comment (mathisscott) [inline packages/core/src/accordion/entrypoint.tsconfig.json]: since accordion is using icons internally, i think we also need the icons path here.

See `./core/src/forms/entrypoint.tsconfig.json`

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+ * @cssprop --header-open-background-color
+ * @cssprop --transition-duration
+ */
+export class CdsAccordionPanel extends CdsBaseAccordionPanel {
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute('slot', 'accordion-panel');
+  }
+
+  private toggle() {
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.ts:63]: How will end users escape hatch this?

File: packages/core/src/accordion/accordion-panel.element.spec.ts [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    );
+  });
+
+  it('should have a slot attribute of value `accordion-panel`', async () => {
+    await componentIsStable(component);
+    expect(component.hasAttribute('slot')).toBe(true);
+    expect(component.getAttribute('slot')).toEqual('accordion-panel');
+  });
+
+  it('should emit an expandedChange event when header is clicked', async done => {
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.spec.ts:42]: The click is also changing whether the panel is expanded or not. I don't know if that's the path we want to go down. But if it is, we should write a unit test to verify that state is managed.

File: packages/core/src/accordion/accordion-panel.element.scss [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+// todo: set max-height for content
+.accordion-content-wrapper {
+  height: 0;
+  transition: transform height var(--transition-duration);
+  transition-timing-function: cubic-bezier(0, 0.99, 0, 0.99);
+
+  overflow-y: hidden;
+
+  &.open {
+    transform: scaleY(1);
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.scss]: is it our intent to keep the transform here? because it will break overlays and we don't seem to be making use of it anywhere else.

File: packages/core/src/accordion/accordion-panel.element.scss [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+    ::slotted([slot='accordion-header']) {
+      cursor: not-allowed;
+    }
+  }
+
+  &.open {
+    --header-background-color: var(--header-open-background-color);
+
+    .accordion-angle-wrapper {
+      transform: rotate(90deg);
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.scss:76]: this seems related to my comment on giving users the option to customize the accordion icon below.

File: packages/core/src/accordion/accordion-header.element.ts [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * Web component accordion
+ *
+ * ```typescript
+ * import '@clr/core/accordion/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-accordion>
+ *   <cds-accordion-panel expanded='panel1Expanded' (expandedChange)='panel1Expanded = $event.detail'>
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-header.element.ts]: Looks like more angular code. i'm not sure we need a full accordion code example here...

File: packages/core/src/accordion/accordion-content.element.ts [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+/**
+ * Web component accordion
+ *
+ * ```typescript
+ * import '@clr/core/accordion/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-accordion>
+ *   <cds-accordion-panel expanded='panel1Expanded' (expandedChange)='panel1Expanded = $event.detail'>
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-content.element.ts]: Same question here re: Angular looking code and whether we need a full accordion code example.

**Review by mathisscott — CHANGES_REQUESTED** [review]

File: packages/core/src/internal/base/accordion-panel.base.ts (hunk 1 above)
Comment (jeeyun) [reply packages/core/src/internal/base/accordion-panel.base.ts]: Thank you for that feedback. Our Angular component actually follows this and stepper extends from accordion (including sub-components). I found it to be a bit hard to follow on the template side, but I don't think we have to keep that the same in core so let me take a look. I think your points are valid and I could probably come up with a decent solution.

File: packages/core/src/accordion/accordion-panel.element.scss [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+  --font-weight: #{$cds-token-typography-font-weight-regular};
+  --header-background-color: #{$cds-token-color-neutral-50};
+  --header-color: #{$cds-token-color-neutral-800};
+  --header-font-size: #{$cds-token-typography-font-size-3};
+  --header-font-weight: #{$cds-token-typography-font-weight-semibold};
+  --header-open-background-color: #{$cds-token-global-selection-color};
+  --transition-duration: 200ms;
+}
+
+::slotted(cds-accordion-header) {
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion-panel.element.scss:17]: A challenge we've had with using `cds-*` element selectors is that the ShadyDOM polyfill will rewrite the elements. Basically turning them into spans with classnames.

For this reason, we have tried to avoid using the custom elements in our CSS selectors. Because we can't count on them being there.

I don't know if this still applies after the IE11 announcement this month. @coryrylan may have more info on how this does or doesn't affect Safari.

But just a heads up that all these custom element selectors may need to change...

File: packages/core/src/accordion/accordion-panel.element.scss (hunk 13 above)
Comment (mathisscott) [reply packages/core/src/accordion/accordion-panel.element.scss:76]: Another issue here is that there is an animation when the panel opens but no animation when it closes. Is that intentional?

File: packages/core/src/accordion/accordion.stories.ts [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+export const customStyles = () => {
+  return html`
+    <style>
+      cds-accordion-panel.app-custom {
+        --border-color: midnightblue;
+        --header-color: aliceblue;
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.stories.ts]: Could we include changing the color of the caret/angle icon here? I think it would be helpful for people to see/know they can do that.

File: packages/core/src/accordion/accordion.stories.mdx [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+## Accordion
+
+<Preview>
+  <Story id="components-accordion-stories--basic-accordion" />
+</Preview>
+
+## Custom accordion styles
+
+<Preview>
+  <Story id="components-alert-stories--custom-styles" />
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.stories.mdx]: Should this point to the accordion stories?

File: packages/core/src/accordion/accordion.stories.mdx [hunk 19]
```diff
@@ -0,0 +1,51 @@
+import { Meta, Props, Story, Preview, API } from '@storybook/addon-docs/blocks';
+import { html, LitElement } from 'lit-element';
+
+<Meta title="Components/Accordion/Getting Started" />
```
Comment (mathisscott) [inline packages/core/src/accordion/accordion.stories.mdx]: The stories for the accordion need to go into the components preview section. New components go through one full version preview.

**Review by mathisscott — CHANGES_REQUESTED** [review]

File: packages/core/src/accordion/accordion.element.ts (hunk 5 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion.element.ts]: it doesn't. removing. 😄

File: packages/core/src/accordion/accordion.element.ts (hunk 6 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion.element.ts]: yeah that's a good point. i think initially we populated the sample codes in Angular and I probably carried it over. I'll update this.

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 7 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.ts:69]: I think those are different use cases and all valid actually: https://stackoverflow.com/questions/61490521/dot-with-html-elements-attribute-property

- boolean attribute: `?disabled=${this.disabled}`

- property: `.disabled=${this.disabled}`

- attribute value: `<div class="${...}></div>`

In this component we are using the boolean attribute on an html button, and the conditional statement for the div class.

In the case of `control.element.ts` it's actually a property (line 105) so it'd be correct to use the `.disabled="${this.disabled}"` there.

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 8 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.ts]: The accordion angle does change direction when open but via transform and not by changing the attribute value. I'll still update to use `direction`. 

Can you tell me what may be missing from a11y point of view? I vetted it against our Angular implementation and also did screenreader testing locally. It's possible I missed something though.

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 7 above)
Comment (mathisscott) [reply packages/core/src/accordion/accordion-panel.element.ts:69]: ah. i see. maybe that's a good candidate for the contribution guidelines if it isn't there already. i'll make a note to check and add that to them if they aren't there. just looking at it, it looks like we can't make up our mind how we want to do it. but understanding it's actually three different things...

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 8 above)
Comment (mathisscott) [reply packages/core/src/accordion/accordion-panel.element.ts]: An `aria-label` on the icon is probably called for...

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 10 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.ts:63]: You are right; this should be the responsibility of the application. Removing the auto toggling from component logic. Thanks for catching that.

File: packages/core/src/accordion/accordion-panel.element.spec.ts (hunk 11 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.spec.ts:42]: Yup. Updating the logic. It should be handled by the component.

File: packages/core/src/accordion/accordion-panel.element.scss (hunk 12 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.scss]: Removing.

File: packages/core/src/internal/base/accordion-panel.base.ts (hunk 1 above)
Comment (jeeyun) [reply packages/core/src/internal/base/accordion-panel.base.ts]: Updated to get rid of the base class!

File: packages/core/src/accordion/accordion-panel.element.scss (hunk 13 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.scss:76]: Fixed to animate on both expand and collapse.

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 8 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.ts]: The icon by itself is not actionable and is somewhat decorative. I've taken a look at couple of other aria accordions as well. Maybe it makes more sense to set `aria-hidden=true` so that it's hidden from SR user?

File: packages/core/src/accordion/accordion-panel.element.scss (hunk 16 above)
Comment (coryrylan) [reply packages/core/src/accordion/accordion-panel.element.scss:17]: With ie11 drop we no longer have to worry about this and can use element selectors 👍

Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.scss:17]: Awesome! 🎉

File: packages/core/src/accordion/accordion-panel.element.scss [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+  --border-width: #{$cds-token-global-border-width};
+  --color: #{$cds-token-color-neutral-800};
+  --icon-color: #{$cds-token-color-neutral-700};
+  --font-size: #{$cds-token-typography-font-size-4};
+  --font-weight: #{$cds-token-typography-font-weight-regular};
+  --header-background-color: #{$cds-token-color-neutral-50};
+  --header-color: #{$cds-token-color-neutral-800};
+  --header-font-size: #{$cds-token-typography-font-size-3};
+  --header-font-weight: #{$cds-token-typography-font-weight-semibold};
+  --header-open-background-color: #{$cds-token-global-selection-color};
```
Comment (coryrylan) [inline packages/core/src/accordion/accordion-panel.element.scss]: May not be worth the effort to adjust templates but we could move these to the `cds-accordion-header` element similar to alerts/alert groups. Then the component doesnt need nested css props on the public API and may not need conditional classes in the component template.

```scss
// cds-accordion-header css
:root {
  --background: #{$cds-token-color-neutral-50};
  --color: #{$cds-token-color-neutral-800};
  --font-size: #{$cds-token-typography-font-size-3};
  --font-weight: #{$cds-token-typography-font-weight-semibold};
}
```

```scss
// cds-accordion-panel css
:host([expanded]) ::slotted(cds-accordion-header) {
  --background: #{$cds-token-global-selection-color};
}
```

```css
cds-accordion-header {
  --background: ...
  --color: ...
  --font-size: ...
  --font-weight: ...
}
```

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+ * @slot
+ * @slot cds-accordion-header
+ * @slot cds-accordion-content
+ * @event expandedChange - notify when the user has clicked the panel header
+ * @cssprop --border-color
+ * @cssprop --border-width
+ * @cssprop --color
+ * @cssprop --font-size
+ * @cssprop --font-weight
+ * @cssprop --header-background-color
```
Comment (coryrylan) [inline packages/core/src/accordion/accordion-panel.element.ts]: nit: using/assigning to `background` instead of `background-color` gives a bit more flexibility with theming

**Review by coryrylan — APPROVED** [review]

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { baseStyles, event, EventEmitter } from '@clr/core/internal';
+import { ClarityIcons, angleIcon } from '@clr/core/icon';
+import { html, LitElement, property } from 'lit-element';
+import { styles } from './accordion-panel.element.css.js';
+
+ClarityIcons.addIcons(angleIcon);
```
Comment (coryrylan) [inline packages/core/src/accordion/accordion-panel.element.ts]: Will want to move this to the `register.ts` so it is not removed by tree shaking

**Review by coryrylan — CHANGES_REQUESTED** [review]

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 22 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.ts]: Ah yes, thank you for catching that. Will move this.

File: packages/core/src/accordion/accordion.stories.ts (hunk 4 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion.stories.ts:45]: As discussed, we'll incorporate this into the pattern work since this is more of a "how to use this component in an application" rather than component's API.

Comment (mathisscott) [reply packages/core/src/accordion/accordion.stories.ts:45]: Because we have no place to put "patterns" and because we have no current plan in place as to how we are tackling patterns, documenting them, or storing them for users to get to, I'm concerned that this is pushing a common ask/use case into the indefinite future. I feel like we're going to get asked this and have to answer it over and over. 

I've erred on the side of including a "pattern" or two in the stories for now (see: accessible circular progress, nested/layered overlays, etc.) so that users have working examples now. Given the limitations of storybook, there's a bit of a leap between the story and what a user would implement. But the story demonstrates that something is possible and the raw mechanics of how it would be accomplished.

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 8 above)
Comment (mathisscott) [reply packages/core/src/accordion/accordion-panel.element.ts]: `aria-hidden="true'` makes a lot of sense in that case...

File: packages/core/src/accordion/accordion-panel.element.scss (hunk 20 above)
Comment (jeeyun) [reply packages/core/src/accordion/accordion-panel.element.scss]: I started out trying to refactor but I think I'll leave it as is for now.

Animation also relies on the conditional css class and some parts of the header template rely on both `expanded` and `disabled` states that are available in the panel, making the refactor a bit less clean-cut than initially anticipated. I'm not opposed to improving things though, so if you think it might still be worth the effort let me know and maybe we can pair on this?

Comment (coryrylan) [reply packages/core/src/accordion/accordion-panel.element.scss]: Going to resolve since I'm not sure if its really worth the effort especially with the new theming coming soon after.

File: packages/core/src/accordion/accordion-panel.element.ts [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+ * @cssprop --transition-duration
+ */
+export class CdsAccordionPanel extends CdsBaseAccordionPanel {
+  connectedCallback() {
+    super.connectedCallback();
+    this.setAttribute('slot', 'accordion-panel');
+  }
+
+  private toggle() {
+    this.expanded = !this.expanded;
```
Comment (coryrylan) [inline packages/core/src/accordion/accordion-panel.element.ts]: Not sure if this was removed based on the comments above but looks like its still needs to be removed to prevent the stateful escape hatch.

File: packages/core/src/accordion/accordion.stories.ts (hunk 4 above)
Comment (coryrylan) [reply packages/core/src/accordion/accordion.stories.ts:45]: I think something like this will work in storybook

```html
<cds-accordion>
  <cds-accordion-panel expanded>
    <cds-accordion-header>Header 1</cds-accordion-header>
    <cds-accordion-content>Content 1</cds-accordion-content>
  </cds-accordion-panel>
  <cds-accordion-panel id="async-panel">
    <cds-accordion-header>Header 1</cds-accordion-header>
    <cds-accordion-content>
      
    </cds-accordion-content>
  </cds-accordion-panel>
</cds-accordion>
<script>
  const asyncPanel = document.querySelector('#async-panel');
  const asyncPanelContent = asyncPanel.querySelector('cds-accordion-content');

  asyncPanel.addEventListener('expandedChange', () => {
    asyncPanel.expanded = !asyncPanel.expanded;
    asyncPanelContent.innerHTML = `<cds-progress-circle></cds-progress-circle>`;

    if (asyncPanel.expanded) {
      setTimeout(() => asyncPanelContent.innerHTML = `<p>hello world</p>`, 3000);
    }
  });
</script>
```

File: packages/core/src/accordion/accordion-panel.element.ts (hunk 23 above)
Comment (coryrylan) [reply packages/core/src/accordion/accordion-panel.element.ts]: nvm I see it :)

File: packages/core/src/internal/utils/metadata.ts [hunk 24]
```diff
@@ -18,7 +18,7 @@ export function getElementStorybookArgs(args: any) {
 
 function getCustomPropsStorybookArgs(customElementMetadata: any) {
   return customElementMetadata.cssProperties?.reduce((obj: any, prop: { name: string }) => {
-    const type = prop?.name?.includes('color') || prop.name === '--background' ? 'color' : 'text';
+    const type = prop?.name?.includes('color') || prop.name?.endsWith('background') ? 'color' : 'text';
```
Comment (mathisscott) [inline packages/core/src/internal/utils/metadata.ts]: wouldn't we need this to be...

```
const type = prop?.name?.includes('color') || prop?.name?.endsWith('background') ? 'color' : 'text';
```

If `prop` can be null the first time around, I don't see why it couldn't be null the second.

**Review by mathisscott — CHANGES_REQUESTED** [review]

**Review by coryrylan — APPROVED** [review]

File: packages/core/src/accordion/accordion.stories.ts (hunk 4 above)
Comment (mathisscott) [reply packages/core/src/accordion/accordion.stories.ts:45]: I had a hard time tracking this down but I see the example in `accordion-panel.stories.ts`

Thanks for doing that!
